exchanges/okx/okx.go

```diff
@@ -0,0 +1,4441 @@
+func (ok *Okx) CancelAlgoOrder(ctx context.Context, args []AlgoOrderCancelParams) ([]AlgoOrder, error) {
+	return ok.cancelAlgoOrder(ctx, args, cancelAlgoOrder)
+}
+
+// cancelAlgoOrder to cancel unfilled algo orders.
+func (ok *Okx) cancelAlgoOrder(ctx context.Context, args []AlgoOrderCancelParams, route string) ([]AlgoOrder, error) {
+	for x := range args {
+		arg := args[x]
+		if arg.AlgoOrderID == "" {
+			return nil, errMssingAlgoOrderID
+		} else if arg.InstrumentID == "" {
+			return nil, errMissingInstrumentID
+		}
+	}
+	if len(args) == 0 {
+		return nil, errors.New("no parameter")
+	}
+	var resp []AlgoOrder
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, cancelAlgoOrderEPL, http.MethodPost, route, args, &resp, true)
+}
+
+// GetAlgoOrderList retrieves a list of untriggered Algo orders under the current account.
+func (ok *Okx) GetAlgoOrderList(ctx context.Context, orderType, algoOrderID, clientSuppliedOrderID, instrumentType, instrumentID string, after, before time.Time, limit int64) ([]AlgoOrderResponse, error) {
+	params := url.Values{}
+	orderType = strings.ToLower(orderType)
+	if !(orderType == "conditional" ||
+		orderType == "oco" ||
+		orderType == "trigger" ||
+		orderType == "move_order_stop" ||
+		orderType == "iceberg" ||
+		orderType == "twap") {
+		return nil, fmt.Errorf("invalid order type value %s,%s,%s,%s,%s,and %s", "conditional", "oco", "trigger", "move_order_stop", "iceberg", "twap")
+	}
+	params.Set("ordType", orderType)
+	var resp []AlgoOrderResponse
+	if algoOrderID != "" {
+		params.Set("algoId", algoOrderID)
+	}
+	if clientSuppliedOrderID != "" {
+		params.Set("clOrdId", clientSuppliedOrderID)
+	}
+	if instrumentType == okxInstTypeSpot ||
+		instrumentType == okxInstTypeSwap ||
+		instrumentType == okxInstTypeFutures ||
+		instrumentType == okxInstTypeOption {
+		params.Set("instType", instrumentType)
+	}
+	if instrumentID != "" {
+		params.Set("instId", instrumentID)
+	}
+	if !before.IsZero() && before.Before(time.Now()) {
+		params.Set("before", strconv.FormatInt(before.UnixMilli(), 10))
+	}
+	if !after.IsZero() && after.After(time.Now()) {
+		params.Set("after", strconv.FormatInt(after.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getAlgoOrderListEPL, http.MethodGet, common.EncodeURLValues(getAlgoOrders, params), nil, &resp, true)
+}
+
+// GetAlgoOrderHistory load a list of all algo orders under the current account in the last 3 months.
+func (ok *Okx) GetAlgoOrderHistory(ctx context.Context, orderType, state, algoOrderID, instrumentType, instrumentID string, after, before time.Time, limit int64) ([]AlgoOrderResponse, error) {
+	params := url.Values{}
+	orderType = strings.ToLower(strings.Trim(orderType, " "))
+	if !(orderType == "conditional" || orderType == "oco" || orderType == "trigger" || orderType == "move_order_stop" ||
+		orderType == "iceberg" || orderType == "twap") {
```

Comment:
```suggestion
	if orderType != "conditional" && orderType != "oco" && orderType != "trigger" && orderType != "move_order_stop" && orderType != "iceberg" && orderType != "twap" {
```
Why not use the consts?